internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_controller.go

```diff
@@ -370,6 +370,17 @@ func (c *Controller) phases(seq runtime.Sequence, data interface{}) ([]runtime.P
 		}
 
 		phases = c.s.StageUpgrade(c.r, in)
+	case runtime.SequenceMaintenanceUpgrade:
+		var (
+			in *machine.UpgradeRequest
+			ok bool
+		)
+
+		if in, ok = data.(*machine.UpgradeRequest); !ok {
+			return nil, runtime.ErrInvalidSequenceData
+		}
```

Comment:
```suggestion
		in, ok := data.(*machine.UpgradeRequest);
		if !ok {
			return nil, runtime.ErrInvalidSequenceData
		}
```

Minor nit: this looks shorter

Reply:
fixed